### PR TITLE
fix: make test data submodule init optional

### PR DESCRIPTION
**Pull request by kayarre**
Make `submodule init` optional so test data can be selected or edited without overwrite during build process.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -102,6 +102,8 @@ option( VMTK_BUILD_TESTING
   "Build the VMTK testing libraries" OFF)
 
 if(VMTK_BUILD_TESTING)
+  option( VMTK_INIT_TEST_DATA
```
Comment (jcfr) [inline CMakeLists.txt:105]: To explicitly notify the developer of locally modified files, what do you think of explicitly checking for locally modified files using approach like this one [used in ExternalProject-gitupdate.cmake.in](https://github.com/Kitware/CMake/blob/065604b2b3cabef95c0f406c990274f439452bbd/Modules/ExternalProject-gitupdate.cmake.in#L152-L166).

If there are modifications, the call to `git submodule update --init` is skipped and a warning is reported.

If the directory is empty (and build testing is enabled), the call to `git submodule update --init` is always performed.

Comment (kayarre) [reply CMakeLists.txt:105]: I am fine with that approach, I had not considered it. It will take me some time to figure out and implement.

Comment (kayarre) [reply CMakeLists.txt:105]: Another thought is that if the test data is explicitly on a different  branch and not  limited to modified repo, we don't want to update it in that instance either. I am not sure the proposed method accounts for this, but will test.

Comment (kayarre) [reply CMakeLists.txt]: I updated things to use what you suggested. I tested that it does not overwrite the data in the unclean state.

**Comment by kayarre** [comment]
@jcfr @lassoan is this good to merge?

**Review by lassoan — APPROVED** [review]
It looks good to me, thank you.

**Pull request merged**